### PR TITLE
I590 Add NormalisedRootMeanSquareError

### LINKED ISSUE #590: Add score function 'RRMSE'?
Sometime when I compare different output where the absolute magnitude does not really matter, I prefer to have a 'relative' RMSE score, which is simply `RMSE divided by RMSD of observed data to zero`. Maybe worth adding to PINTS?

### PR DESCRIPTION
**Pull request by chonlei**
Closes #590.

I have named it as 'NormalisedRootMeanSquareError', as it does seem like we are normalising the RMSE.

### CODE REVIEW

File: docs/source/error_measures.rst [hunk 1]
```diff
@@ -23,6 +23,8 @@ Example::
 
 .. autoclass:: RootMeanSquaredError
 
+.. autoclass:: NormalisedRootMeanSquaredError
```
Comment (MichaelClerx) [inline docs/source/error_measures.rst:26]: We're doing everything alphabetical now, so this needs to go up before ProbabilityBasedError !

File: pints/__init__.py [hunk 2]
```diff
@@ -140,6 +140,7 @@ def version(formatted=False):
     SumOfErrors,
     MeanSquaredError,
     RootMeanSquaredError,
+    NormalisedRootMeanSquaredError,
```
Comment (MichaelClerx) [inline pints/__init__.py:143]: Same here

File: pints/__init__.py [hunk 3]
```diff
@@ -140,6 +140,7 @@ def version(formatted=False):
     SumOfErrors,
```
Comment (MichaelClerx) [inline pints/__init__.py:140]: This one's also wrong :-D

File: pints/_error_measures.py [hunk 4]
```diff
@@ -272,6 +272,35 @@ def __call__(self, x):
             (self._problem.evaluate(x) - self._values)**2))
 
 
+class NormalisedRootMeanSquaredError(ProblemErrorMeasure):
```
Comment (MichaelClerx) [inline pints/_error_measures.py:275]: Again, alphabetical!

The reason is, when there's only a few classes you can make a sensible order, but then when there's more it becomes less clear and different people have different opinions so it's easier to just do everything alphabetical. Also makes it easier to find stuff :-)

File: pints/_error_measures.py [hunk 5]
```diff
@@ -272,6 +272,35 @@ def __call__(self, x):
...
+    """
+    Calculates a normalised root mean squared error (RMSE):
```
Comment (MichaelClerx) [inline pints/_error_measures.py:277]: This makes me think (normalised root mean square error) = (RMSE)
Maybe clearer to write "normalised RMSE (root mean square error)" in this case?

You might even want to break it up into multiple sentences, e.g.

```
"""
Calculates a normalised root mean square error.

This is similar to the (unnormalised) root mean square error (RMSE) defined as
``...``
but
...
```
?

Comment (MichaelClerx) [reply pints/_error_measures.py:277]: Also would it be clearer to define it as ``sqrt(....) / N`` ? (even though in code it's better the way you have it)

File: pints/_error_measures.py [hunk 6]
```diff
@@ -272,6 +272,35 @@ def __call__(self, x):
...
+    ``f = N * sqrt( sum( (x[i] - y[i])**2 / n) )``
+    where ``N`` is a normalisation constant defined as the inverse of the RMSE
+    of the data ``y`` with respect to zero.
```
Comment (MichaelClerx) [inline pints/_error_measures.py:280]: Is the error-w.r.t.-zero way the best way to describe it? Is it conceptually an error w.r.t. a zero signal? Or do you derive the measure by thinking of sqrt(sum(y**2)) as some kind of magnitude?

File: pints/tests/test_error_measures.py [hunk 7]
```diff
@@ -292,6 +292,25 @@ def test_root_mean_squared_error(self):
...
+        self.assertNotEqual(np.all(e([1, 2, 3])), 0)
+        x = [0, 0, 0]
+        y = 1.0
+        self.assertEqual(e(x), y)
+        x = [1, 1, 1]
+        y = np.sqrt((4 + 1 + 4) / 3) / np.sqrt((1 + 4 + 9) / 3)
+        self.assertAlmostEqual(e(x), y)
+
+        p = MultiMiniProblem()
+        self.assertRaises(ValueError, pints.NormalisedRootMeanSquaredError, p)
```
Comment (MichaelClerx) [inline pints/tests/test_error_measures.py:312]: Can we do an assertRaisesRegex here?

File: docs/source/error_measures.rst (hunk 1 above)
Comment (chonlei) [reply docs/source/error_measures.rst:26]: :D OK! I thought about it, but then it didn't look alpahbetically ordered, so I put it to where I thought the most sensible place. Sure, I can move it!

File: pints/__init__.py (hunk 2 above)
Comment (chonlei) [reply pints/__init__.py:143]: But SumofErrors is before MeanSquaredError... Which bit is alphabetically ordered?

File: pints/__init__.py (hunk 3 above)
Comment (chonlei) [reply pints/__init__.py:140]: Oh! OK, you just answered my question above. I will move it too then.

Comment (chonlei) [reply pints/__init__.py:140]: Is it just for the docs/__init__.py only? Or I need to move it for the actual _error_measures.py functions as well...?!

File: pints/_error_measures.py (hunk 4 above)
Comment (chonlei) [reply pints/_error_measures.py:275]: You kept answering my questions above! :-D

File: pints/_error_measures.py (hunk 6 above)
Comment (chonlei) [reply pints/_error_measures.py:280]: Do you have any suggestion on how to word it?

File: pints/__init__.py (hunk 2 above)
Comment (MichaelClerx) [reply pints/__init__.py:143]: They should have been!
You can fix all of them :D

File: pints/__init__.py (hunk 3 above)
Comment (MichaelClerx) [reply pints/__init__.py:140]: docs, imports, and actual code!

Comment (MichaelClerx) [reply pints/__init__.py:140]: We've done it in _most_, but not all places, so you might see some other instances where it's wrong

File: pints/_error_measures.py (hunk 6 above)
Comment (MichaelClerx) [reply pints/_error_measures.py:280]: Maybe just write out the equation in full so you don't need to use too many words?

File: pints/_error_measures.py (hunk 4 above)
Comment (chonlei) [reply pints/_error_measures.py:275]: I have moved this one up too. But there're a few classes that others inherent, they can't go in alphabetical order. So I have left the others stay unchanged.

File: pints/_error_measures.py (hunk 5 above)
Comment (chonlei) [reply pints/_error_measures.py:277]: Done!

File: pints/tests/test_error_measures.py (hunk 7 above)
Comment (chonlei) [reply pints/tests/test_error_measures.py]: Yup -- changed it for the RootMeanSqauredError unit test too!

**Pull request merged**